tf2_ros/include/tf2_ros/buffer_client.h

```diff
@@ -126,6 +126,7 @@ namespace tf2_ros
       {
         client_ = rclcpp_action::create_client<LookupTransformAction>(node, ns);
       }
+      virtual ~BufferClient() = default;
```

Comment:
I definitely agree that this is a problem.  However, I'm actually more inclined to make `BufferClient` final and just not bother with the virtual destructor at all.  Looking through all of the open-source Kinetic source code, there isn't a single instance of anyone subclassing it.  So I think it is somewhat safe to say that this is almost always used as a "has-a" class, rather than an "is-a".  @tfoote do you have any thoughts here?

Reply:
Either would be ok with me. `final` makes classes harder to use with mocking frameworks though, I don't know if we feel it could be a concern in this case.

Reply:
I'm going to run CI on it as-is.  It's a little safer this way.

Reply:
Thanks!